Classes/Model/BasePDF.php

```diff
@@ -196,11 +201,16 @@ public function AddPage($orientation = '', $format = '', $rotationOrKeepmargins
             $this->renderHeader();
         }
 
-        if ($this->importTemplateOnThisPage && $this->tpl !== 0) {
-            $this->useTemplate($this->tpl);
+        if ($this->importTemplateOnThisPage && $this->currentTemplateId) {
+            $this->useTemplate($this->currentTemplateId);
         }
     }
 
+    public function setCurrentTemplateId(int $templateId)
```

Comment:
Please do not use type hints for primitive data types as we still support PHP 5.6.

Reply:
> we still support PHP 5.6

I guess only for December, right? ;-)

Reply:
We are thinking about dropping support for PHP < 7 next year, yes. But if we do so, we would also like to add type hints to all functions and that would be considered a breaking change. Thus I am not yet sure how fast we will make this transition.